pkg/k8sops/m3db/statefulset_test.go

```diff
@@ -181,3 +184,69 @@ func TestGenerateStatefulSetAffinity(t *testing.T) {
 		assert.Equal(t, expTerms, terms[0].MatchExpressions)
 	}
 }
+
+func TestGenerateStatefulSetPodAntiAffinity(t *testing.T) {
+	tests := []struct {
+		isoGroup myspec.IsolationGroup
+		expBool  bool
+		expErr   error
+	}{
+		{
+			isoGroup: myspec.IsolationGroup{
+				Name: "group1",
+			},
+			expBool: false,
+		},
+		{
+			isoGroup: myspec.IsolationGroup{
+				Name:               "group2",
+				UsePodAntiAffinity: false,
+			},
+			expBool: false,
+		},
+		{
+			isoGroup: myspec.IsolationGroup{
+				Name:                  "group3",
+				UsePodAntiAffinity:    true,
+				PodAffinityToplogyKey: "hostname",
+			},
+			expBool: true,
+		},
+		{
+			isoGroup: myspec.IsolationGroup{
+				Name:               "group4",
+				UsePodAntiAffinity: true,
+			},
+			expBool: true,
+			expErr:  errEmptyPodAffinityToplogyKey,
+		},
+	}
+
+	for _, test := range tests {
+		antiaffinity, err := GenerateStatefulSetPodAntiAffinity(test.isoGroup)
+
+		if !test.expBool {
+			assert.Nil(t, antiaffinity)
+			continue
+		}
+
+		if test.expErr != nil {
+			assert.Equal(t, test.expErr, err)
+			continue
+		}
+
+		terms := antiaffinity.RequiredDuringSchedulingIgnoredDuringExecution
+
+		expTerms := &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      labels.Component,
+					Operator: "In",
+					Values:   []string{labels.ComponentM3DBNode},
+				},
+			},
+		}
+
+		assert.Equal(t, expTerms, terms[0].LabelSelector)
```

Comment:
Can you also verify that `PodAffinityToplogyKey` is set to the expected value in the tests?